src/platform/backends/lxd/lxd_vm_image_vault.cpp

```diff
@@ -225,13 +195,64 @@ bool mp::LXDVMImageVault::has_record_for(const std::string& name)
 
 void mp::LXDVMImageVault::prune_expired_images()
 {
-    mpl::log(mpl::Level::trace, category, "Pruning expired images not implemented");
+    auto json_reply = lxd_request(manager, "GET", QUrl(QString("%1/images").arg(base_url.toString())));
+
+    auto images = json_reply["metadata"].toArray();
+
+    for (const auto image : images)
+    {
+        auto image_info = image.toObject();
+        auto last_used_count =
+            QDateTime::fromString(image_info["last_used_at"].toString(), Qt::ISODateWithMs).toMSecsSinceEpoch();
+        auto last_used = std::chrono::system_clock::time_point(std::chrono::milliseconds(last_used_count));
+
+        if (image_info.contains("update_source") && last_used + days_to_expire <= std::chrono::system_clock::now())
+        {
+            mpl::log(mpl::Level::info, category,
+                     fmt::format("Source image \'{}\' is expired. Removing it…",
+                                 image_info["properties"].toObject()["release"].toString()));
+
+            lxd_request(
+                manager, "DELETE",
+                QUrl(QString("%1/images/%2").arg(base_url.toString()).arg(image_info["fingerprint"].toString())));
+        }
+    }
 }
 
 void mp::LXDVMImageVault::update_images(const FetchType& fetch_type, const PrepareAction& prepare,
                                         const ProgressMonitor& monitor)
 {
-    mpl::log(mpl::Level::trace, category, "Updating images not implemented");
+    auto json_reply = lxd_request(manager, "GET", QUrl(QString("%1/images").arg(base_url.toString())));
+
+    auto images = json_reply["metadata"].toArray();
+
+    for (const auto image : images)
+    {
+        auto image_info = image.toObject();
+        if (image_info.contains("update_source"))
```

Comment:
Shouldn't we check that it has an alias? Does asking for a refresh make sense otherwise?

Reply:
From what I can tell, this is how LXD stores image info when an image has been download via an alias, so we only want to update these types of images and not update images that have been launched via a hash.

Reply:
Sure, I understand what you want to achieve, but can't you differentiate based on `alias` being empty instead? I feel like `update_source` could very well be filled for hash-sourced images, and relying on that could be dangerous.

Reply:
`alias` is empty no matter what and `update_source` only exists if the image was downloaded via an alias.  This was kind of confusing to me when developing this, but I figure it is because how we are requesting an image to be downloaded outside of LXD's launch handling.  I did not follow up with the LXD folks about this.

Reply:
ACK